person.rb

```diff
@@ -0,0 +1,28 @@
+require_relative 'nameable'
+
+class Person < Nameable
+  def initialize(id, age, name = 'Unknown', parent_permission: true)
+    @id = id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    super()
+  end
+
+  attr_accessor :name, :age
+  attr_reader :id
```

Comment:
- [ ] Great effort working on your decorator, But when i try testing the decorator using the `person.rb` file , I notice the decorator is throwing an uninitialized constant error, So i would kindly suggest your require the decorator file in the `person.rb` file and i believe this will solve the uninitialized error

E.g
```
require_relative 'decorator'
```